sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -132,17 +174,30 @@ class SQLContext(@transient val sparkContext: SparkContext)
   protected[sql] lazy val optimizer: Optimizer = DefaultOptimizer
 
   @transient
-  protected[sql] val ddlParser = new DDLParser(sqlParser.parse(_))
+  protected[sql] val ddlParser = new DDLParser((sql: String) => { getSQLDialect().parse(sql) })
 
   @transient
-  protected[sql] val sqlParser = {
-    val fallback = new catalyst.SqlParser
-    new SparkSQLParser(fallback.parse(_))
+  protected[sql] val sqlParser = new SparkSQLParser(getSQLDialect().parse(_))
```

Comment:
I think we'd better keep it, not just for the mima test, but also for the sub class of `Dialect`. e.g. we have to specify the `SparkSQLParser` for `HiveQLDialect`.